homeassistant/components/sonos/speaker.py

```diff
@@ -612,22 +606,18 @@ async def async_poll_battery(self, now: datetime.datetime | None = None) -> None
     #
     # Group management
     #
-    def update_groups(self, event: SonosEvent | None = None) -> None:
-        """Handle callback for topology change event."""
-        coro = self.create_update_groups_coro(event)
-        if coro:
-            self.hass.add_job(coro)  # type: ignore
+    def update_groups(self) -> None:
+        """Update group topology when polling."""
+        self.hass.add_job(self.create_update_groups_coro())
 
     @callback
-    def async_update_groups(self, event: SonosEvent | None = None) -> None:
+    def async_update_groups(self, event: SonosEvent) -> None:
         """Handle callback for topology change event."""
-        coro = self.create_update_groups_coro(event)
-        if coro:
-            self.hass.async_add_job(coro)  # type: ignore
+        if not hasattr(event, "zone_player_uui_ds_in_group"):
+            return None
+        self.hass.async_add_job(self.create_update_groups_coro(event))
```

Comment:
Please use `self.hass.async_create_task`. `hass.async_add_job` is legacy for general use.

Reply:
We're missing an explicit return value here or we should just `return` above.

Reply:
Addressed in #54542.